env/git.go

```diff
@@ -112,26 +112,21 @@ func GitSHA(path string) (string, error) {
 }
 
 var GitBlob = func(path string, commit *object.Commit) (string, error) {
-	if commit == nil {
-		blob, err := fallbackBlob(path)
+	if commit != nil {
+		if file, err := commit.File(path); err == nil {
```

Comment:
Yes, that works for me: if we don't know the commit, and if we don't know the file, then we fallback. This is fine with me and doesn't count as as masking "any error" that I'd be concerned with masking.